ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java

```diff
@@ -185,16 +185,38 @@ public MaterializedWindowedTable windowed() {
     public Iterator<WindowedRow> get(final int partition, final Range<Instant> windowStartBounds,
         final Range<Instant> windowEndBounds) {
       final Iterator<WindowedRow> result = table.get(partition, windowStartBounds, windowEndBounds);
-
       return Streams.stream(result)
           .map(row ->  {
-            return filterAndTransform(row.windowedKey(), row.value(), row.rowTime())
+            return filterAndTransform(row.windowedKey(), getIntermediateRow(row), row.rowTime())
                 .map(v -> row.withValue(v, schema()));
           })
           .filter(Optional::isPresent)
           .map(Optional::get)
           .iterator();
     }
   }
+
+  public static GenericRow getIntermediateRow(final TableRow row) {
```

Comment:
If we don't always want to unconditionally add these fields in the future, we should add a comment saying that we need to address this in the future to decide when we need to add these additional fields.